File: permutations-ii/permutations-ii.cpp
class Solution {
public:
    vector<vector<int> >m;
    set<vector<int> >s;
    void solve(vector<int>nums,int i)
    {
        if(i==nums.size())
        {
            if(s.find(nums)==s.end())
            {
                s.insert(nums);
                m.push_back(nums);
                
            }
            return ;
        }
        for(int j=i;j<nums.size();j++)
        {
            swap(nums[i],nums[j]);
            solve(nums,i+1);
            swap(nums[i],nums[j]);
        }
    }
    vector<vector<int>> permuteUnique(vector<int>& nums) {
        sort(nums.begin(),nums.end());
        solve(nums,0);
        return m;
        
    }
};

File: 47-permutations-ii/permutations-ii.py
class Solution:
    def permuteUnique(self, nums: List[int]) -> List[List[int]]:
        nums = sorted(nums)
        ans = []
        def solve(ind):
            if ind== len(nums):
                ans.append(list(nums))
                return 
            used = set()
            for i in range(ind, len(nums)):
                if nums[i] in used:
                    continue
                used.add(nums[i])
                nums[i], nums[ind] = nums[ind], nums[i]
                solve(ind+1)
                nums[i], nums[ind] = nums[ind], nums[i]
                
        solve(0)
        return ans